src/block.ts

```diff
@@ -1,15 +1,18 @@
-import { BlockHeader } from "./blockHeader";
+import {BlockHeader} from "./blockHeader";
+import { BlockType } from "./blockType";
 import { IModel } from "./model";
 import { Transaction } from "./transaction";
 
 /**
  * Default block in the chain
  */
 export class Block implements IModel {
+  public blockType: BlockType;
   public blockHeader: BlockHeader;
   public transactions: Set<Transaction>;
 
-  constructor(blockHeader: BlockHeader, transactions: Set<Transaction>) {
+  constructor(blockHeader: BlockHeader, transactions: Set<Transaction>, blockType?: BlockType) {
+    this.blockType = blockType || BlockType.REGUALAR;
```

Comment:
`REGUALAR` ?????

Reply:
Please consider changing the optional parameter to a default paramater value.

i.e.
```
constructor(blockHeader: BlockHeader, transactions: Set<Transaction>, blockType: BlockType = BlockType.REGUALAR) {
    this.blockType = blockType;
```

Reply:
I thought we discussed to make it nullable and resolve it to the default value: REGULAR.
This way a block gets the type GENISIS if explicitly stated. This way fewer parameters need to be passed in the constructor.